File: streamx-spark/streamx-spark-monitor/streamx-spark-monitor-core/src/main/java/com/streamxhub/spark/monitor/BootStrap.java
package com.streamxhub.spark.monitor;

import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.transaction.annotation.EnableTransactionManagement;

@SpringBootApplication
@EnableTransactionManagement
@EnableScheduling
@EnableAsync
public class BootStrap {

    public static void main(String[] args) {
        new SpringApplicationBuilder(BootStrap.class).run(args);
    }
}

File: streamx-spark/streamx-spark-monitor/streamx-spark-monitor-core/src/main/scala/com/streamxhub/spark/monitor/core/BootStrap.scala
package com.streamxhub.spark.monitor.core

import org.springframework.context.annotation.Configuration
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.boot.autoconfigure.SpringBootApplication

import org.springframework.boot.SpringApplication

@Configuration
@EnableAutoConfiguration
@ComponentScan(value = Array("com.streamxhub.spark.monitor"))
@SpringBootApplication
class BootStrap

object BootStrap extends App {
  SpringApplication.run(classOf[BootStrap], args: _*)
}
